### PR TITLE
Catch exceptions when parsing invalid XML and return an empty TestFra…

### PR DESCRIPTION
**Pull request by teajay-fr**
Prevent xml errors to block out all test detections.

### CODE REVIEW

**Comment by guwirth** [comment]
Hello @teajay-fr thanks a lot for providing this One point: is there a possibility to write a unit test for this? otherwise :+1: 
@netspiri please review and rate also

**Comment by netspiri** [comment]
Thank you @teajay-fr for your contribution. Much appreciated; especially the fact that you are contributing to the external test runner section.

Due to recent changes in the discovery process, this approach is safer since discovery is halted for a single source and not all of them. Will update other discoverers to utilise a similar approach.

Just a minor comment. Ideally the exception is not silenced but it is logged so that it is visible to end users.

:+1:

File: BoostTestAdapter/Discoverers/ExternalDiscoverer.cs [hunk 1]
```diff
@@ -175,11 +176,17 @@ private TestFramework ParseStaticTestList(string source)
...
-                XmlSerializer deserializer = new XmlSerializer(typeof(TestFramework));
-                return deserializer.Deserialize(stream) as TestFramework;
+                using (FileStream stream = File.OpenRead(path))
+                {
+                    XmlSerializer deserializer = new XmlSerializer(typeof(TestFramework));
+                    return deserializer.Deserialize(stream) as TestFramework;
+                }
             }
+            catch(InvalidOperationException)
+            { }
```
Comment (guwirth) [inline BoostTestAdapter/Discoverers/ExternalDiscoverer.cs]: Add logging please

**Comment by teajay-fr** [comment]
Sure, I can provide a unit test and add some logging, if you didn't do it already. I'll have a look on Monday.

**Comment by teajay-fr** [comment]
Let me know if there is something missing.

**Comment by netspiri** [comment]
:+1:

**Comment by netspiri** [comment]
Noted that the file `BoostTestAdapterNunit/ExternalBoostTestDiscovererTest.cs` is not being included in the unit test module... my bad. Will update `BoostTestAdapterNunit.csproj`. Upcoming rebase should follow this change.

Once again, many thanks for your contributions.

**Pull request merged**